backend/src/sync.js

```diff
@@ -83,6 +83,7 @@ async function saveBlocks(blocksInfo) {
                       signerPublicKey: tx.signer_public_key || tx.public_key,
                       signature: tx.signature,
                       receiverId: tx.receiver_id,
+                      txHeight: blockInfo.header.height + tx_index,
```

Comment:
I am sorry, but this is an invalid assumption. If the current block height is 10, and it has 2 transactions, then we will have transactions with txHeight 10 and 11, but then once the next block comes and has at least one transaction, we have a collision of txHeight 11.